### PR TITLE
Add cardano-wallet, cardano-addresses, cardano-transactions

### PR DESCRIPTION
**Pull request by rvl**
### Overview

- Adds Adrestia repos to the cabal project
- Adds a buildkite pipeline (nightly only) FWIW
- Small random fixes

### Comments

- The cabal configure "Resolving dependencies..." step seems to take noticeably longer after adding these new packages.
- I'm not yet sure how much actually builds, and anyway don't expect the latest master branch of everything to always work together.
- [Buildkite test builds](https://buildkite.com/input-output-hk/cardano-haskell-nightly/builds?branch=rvl%2Fadrestia)

### CODE REVIEW

**Comment by hasufell** [comment]
Don't want to go on a tangent here, but wondering why we don't have a top-level stack.yaml?

**Comment by rvl** [comment]
@hasufell, who are you trying to troll with that comment?

**Comment by hasufell** [comment]
> @hasufell, who are you trying to troll with that comment?

What kind of comment is that? Maybe I'm misunderstanding the purpose of this repo, in which case you could have explained my incorrect assumptions instead of calling out someone a troll, who took the time to look at your PR.

File: cabal.project [hunk 1]
```diff
@@ -48,7 +50,11 @@ packages:
   cardano-db-sync/cardano-db-sync
   cardano-db-sync/cardano-db-sync-extended
 
-index-state: 2020-07-14T22:50:55Z
+  cardano-wallet/lib/*/*.cabal
+  cardano-addresses/*/*.cabal
+  cardano-transactions
```
Comment (KtorZ) [inline cabal.project:55]: It is supposed to be the other way around. cardano-addresses, cardano-wallet & cardano-transactions are using cardano-haskell as a snapshot reference, so they can't be in there.

Comment (rvl) [reply cabal.project:55]: This PR is for adding our stuff to the `cardano-haskell/cabal.project` mega-project. It's a big list of repos which you can (try to) build using a single `cabal build all` command.  

The stack snapshots are different thing. And yes, you are correct that `cardano-{addresses,wallet,transactions}` should not be listed in the stack snapshots.

**Comment by rvl** [comment]
@newhoggy Would you be able to review? Ignore the changes in the `snapshots` subdir. They are from PR #33 being merged in.

File: build.sh [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  cabal_args+=("--store-dir=$CABAL_STORE_DIR")
+fi
+
+################################################################################
+# Caching
+
+restore_cache() {
+  test -n "$cache_args" || exit 0
+  echo "--- Restoring from cache..."
+  cabal-cache sync-from-archive "${cache_args[@]}" || true
```
Comment (newhoggy) [inline build.sh:23]: What's the reason for the `|| true`.  Is this expected to fail under some circumstances and ignored?

**Review by newhoggy — APPROVED** [review]

File: build.sh (hunk 2 above)
Comment (rvl) [reply build.sh:23]: Yeah I wouldn't expect it to fail, but don't want any cache sync problems to stop a build.

**Pull request merged**